services/configuration/src/transactions.rs

```diff
@@ -410,3 +411,35 @@ impl Transaction for VoteAgainst {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use exonum_testkit::{TestKit, TestKitBuilder};
+
+    use super::{Hash, VotingDecisionRef};
+    use errors::Error as ServiceError;
+    use tests::{new_tx_config_vote, new_tx_config_vote_against};
+    use Service as ConfigurationService;
+
+    #[test]
+    fn test_vote_without_propose() {
```

Comment:
Nit: I'd prefer a more "high-level" test, with transactions being sent to the blockchain via `create_block*` and their status checked via explorer; this could be more stable across possible future revisions of transaction logic.

Reply:
I did this test low-level on purpose: high-level test will show that _something_ has broken, and this test in case of any regression will show the exact problem.
IMHO in that particular case the low-level test is better (although I can be wrong).

Reply:
I won't insist. Just a note: It's possible to find the exact error cause via the explorer API (or by looking at the transaction result directly) with something like

```rust
let bogus_tx = ...;
let block = testkit.create_block_with_transaction(tx);
assert_eq!(
    block[0].status().unwrap_err().error_type(),
    TransactionErrorType::Code(ErrorCode::UnknownConfigRef as u8),
);
```